src/core/qgsmaplayer.h

```diff
@@ -1571,6 +1573,15 @@ class CORE_EXPORT QgsMapLayer : public QObject
      */
     void setMapTipTemplate( const QString &mapTipTemplate );
 
+    /**
+     * Returns a provider read flaf deduced from a layer read flags \a layerReadFlags and a dom node \a layerNode
+     * that describing a layer (corresponding to ``maplayer'' tag in a DOM document project file read by QgsProject).
+     * This static method is used when loading a project.
+     *
+     * \since QGIS 3.32
+     */
+    static QgsDataProvider::ReadFlags providerReadFlags( const QDomNode &layerNode, QgsMapLayer::ReadFlags layerReadFlags );
```

Comment:
can we make this protected?

Reply:
no, it used when loading project by QgsProject instance